### PR TITLE
Mongoose data base setup

### PR DESCRIPTION
**Pull request by westonmiller**
### work in progress
  -Adding tests

### What it does?
  -Adds mongoose to package.json
  -Creates database module
    `module.init()` to connect
    `module.User()` to return an User model.
  -Closes database at `SIGINT`

### What will it do?
  -Add test
  -Add documentation to readme  and remove comments from app.js

### Dependencies
  -Run `npm install`
  -Install MongoDB on your local machine

### Links
[Installing MongoDB on mac](https://docs.mongodb.com/manual/tutorial/install-mongodb-on-os-x/#install-mongodb-community-edition) I used manual install as I don't have brew

### Gif

giphy-4

### CODE REVIEW

File: bot/botDatabase.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    "use strict";
+
+    if (modelName === undefined) {
+      modelName = 'User';
+    }
+
+    if (!this.isConnectionOpen) {
+      console.error('!!!!! The Database hasn\'t been initailitzed.  run init() first');
+    }
+    var userSchema = new mongoose.Schema({
```
Comment (reedsa) [inline bot/botDatabase.js]: Would it make sense to refactor the Schema and Model declarations below into a separate file in order to simplify the code here?

File: spec/bodDatabase.spec.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      });
+    });
+  });
+  describe('Updating a User', function() {
+    it('updates a users default location', function(done){
+      var sarah = new User({
+        name: 'sarah'
+      });
+
+      sarah.save(function(error, user){
```
Comment (reedsa) [inline spec/bodDatabase.spec.js]: Should this be changed to 'findOneAndUpdate' instead of 'save'?

File: bot/botDatabase.js (hunk 1 above)
Comment (westonmiller) [inline bot/botDatabase.js]: I think we should at some point after MVP

File: spec/bodDatabase.spec.js (hunk 2 above)
Comment (westonmiller) [inline spec/bodDatabase.spec.js]: fixed.

**Pull request merged**